### PR TITLE
Screen reader and keyboard accessibility (Potential PR to original repo)

### PR DESCRIPTION
**Pull request by hanstdam**
As you can see, this PR contains a few commits, which is not mine. That's because I based my branch on [Eonasdan/bootstrap-datetimepicker development branch](https://github.com/Eonasdan/bootstrap-datetimepicker/tree/development). The [guidelines for contributing](https://github.com/eXplorance/bootstrap-datetimepicker/blob/explorance-master/CONTRIBUTING.md) states that all PR's should be base in that branch, and since I want to create a PR for the main repo, I've included those commits here.

[Here's a link where you can see my changes in isolation.](https://github.com/eXplorance/bootstrap-datetimepicker/compare/3757b3179124adb9e3a982577f85502c77369f7a...a160fa661fa94296de845786d03c37ad9182dfaf)

This PR includes a lot of changes, including 1 (arguably 2) breaking change(s). ~~The `options.keyBinds`, has been been renamed to `options.keyBindings` and expanded with context dependent keyboard navigation functions. I renamed the option to make it clear that a change has been introduced.~~

Other than that, this PR shifts the focus from always being in the input field, to go into the date picker itself. That's in order for the screen reader to read out relevant info based on the users focus. Further several recommendations from [https://www.w3.org/TR/wai-aria-practices/#datepicker](https://www.w3.org/TR/wai-aria-practices/#datepicker) has been implemented.

Check out a demo of the new features here:
https://datetimepicker-demo.neocities.org/

Let me know what you think.

### CODE REVIEW

File: src/js/bootstrap-datetimepicker.js [hunk 1]
```diff
@@ -122,6 +122,10 @@
                 18: 'alt',
                 'space': 32,
                 32: 'space',
+                'end': 35,
+                35: 'end',
+                'home': 36,
+                36: 'home',
```
Comment (pmairoldi) [inline src/js/bootstrap-datetimepicker.js:128]: What does end and home do?

File: src/js/bootstrap-datetimepicker.js [hunk 2]
```diff
@@ -925,6 +1153,11 @@
                 widget.off('click', '[data-action]');
                 widget.off('mousedown', false);
 
+                widget.find('[tabindex="0"]').off({
+                    'keydown': keydown,
+                    'keyup': keyup
+                });
```
Comment (pmairoldi) [inline src/js/bootstrap-datetimepicker.js:1159]: Is this to toggle between the date and time sections?

File: src/js/bootstrap-datetimepicker.js [hunk 3]
```diff
@@ -2125,12 +2439,12 @@
             return picker;
         };
 
-        picker.keyBinds = function (keyBinds) {
+        picker.keyBindings = function (keyBinds) {
```
Comment (pmairoldi) [inline src/js/bootstrap-datetimepicker.js]: I dont think it is a good idea to rename key binds to keyBindings just because it is a breaking change.

**Review by pmairoldi — CHANGES_REQUESTED** [review]
Aside from the keybind name change everything looks good. Maybe some tests could be added for these new things since there are a lot of changes.

File: src/js/bootstrap-datetimepicker.js (hunk 1 above)
Comment (hanstdam) [reply src/js/bootstrap-datetimepicker.js:128]: https://www.w3.org/TR/wai-aria-practices/#datepicker suggests that the `end` button _Moves to the last day of the current month._ and `home` button _Moves to the first day of the current month._.
I didn't add those as defaults, as they did not exist before.

File: src/js/bootstrap-datetimepicker.js (hunk 2 above)
Comment (hanstdam) [reply src/js/bootstrap-datetimepicker.js:1159]: Yes, among others. (I'm assuming you are talking about the events on `[tabindex="0"]` elements in general, and not the removing of events in the `hide` function, right?). If you have toolbars enabled (if `sideBySide: false`, `showTodayButton: true`, `showClear: true` or `showClose: true`), you can `tab` to the different buttons as well, giving you different context, which the screen reader will tell you about.

File: src/js/bootstrap-datetimepicker.js (hunk 3 above)
Comment (hanstdam) [reply src/js/bootstrap-datetimepicker.js]: I can change it back, for sure. No biggie.

**Comment by hanstdam** [comment]
@petester42, Sure, I'll add some tests. Thanks for the feedback! 😄

File: src/js/bootstrap-datetimepicker.js (hunk 2 above)
Comment (pmairoldi) [reply src/js/bootstrap-datetimepicker.js:1159]: ah ok.

File: src/js/bootstrap-datetimepicker.js (hunk 1 above)
Comment (pmairoldi) [reply src/js/bootstrap-datetimepicker.js:128]: If it is best practice for accessibility then why not include the functions?

Comment (hanstdam) [reply src/js/bootstrap-datetimepicker.js:128]: You're right. I'll include them.

Comment (hanstdam) [reply src/js/bootstrap-datetimepicker.js:128]: Now included

**Comment by hanstdam** [comment]
Tests added and all changes implemented based on review.

**Review by pmairoldi — APPROVED** [review]

**Pull request merged**